CreateService/handler.ts

```diff
@@ -0,0 +1,191 @@
+  IResponseErrorNotFound,
+  IResponseErrorTooManyRequests,
+  IResponseSuccessJson,
+  ResponseSuccessJson
+} from "italia-ts-commons/lib/responses";
+
+import {
+  checkSourceIpForHandler,
+  clientIPAndCidrTuple as ipTuple
+} from "io-functions-commons/dist/src/utils/source_ip_check";
+
+import { Context } from "@azure/functions";
+import { identity } from "fp-ts/lib/function";
+import { TaskEither } from "fp-ts/lib/TaskEither";
+import { ServiceModel } from "io-functions-commons/dist/src/models/service";
+import { ContextMiddleware } from "io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import { RequiredBodyPayloadMiddleware } from "io-functions-commons/dist/src/utils/middlewares/required_body_payload";
+import {
+  ObjectIdGenerator,
+  ulidGenerator
+} from "io-functions-commons/dist/src/utils/strings";
+import {
+  EmailString,
+  FiscalCode,
+  NonEmptyString
+} from "italia-ts-commons/lib/strings";
+import { Service } from "../generated/api-admin/Service";
+import { Subscription } from "../generated/api-admin/Subscription";
+import { ServicePayload } from "../generated/definitions/ServicePayload";
+import { ServiceWithSubscriptionKeys } from "../generated/definitions/ServiceWithSubscriptionKeys";
+import { withApiRequestWrapper } from "../utils/api";
+import { APIClient } from "../utils/clients/admin";
+import { getLogger, ILogger } from "../utils/logging";
+import { ErrorResponses, IResponseErrorUnauthorized } from "../utils/responses";
+
+type ResponseTypes =
+  | IResponseSuccessJson<ServiceWithSubscriptionKeys>
+  | IResponseErrorUnauthorized
+  | IResponseErrorForbiddenNotAuthorized
+  | IResponseErrorNotFound
+  | IResponseErrorTooManyRequests
+  | IResponseErrorInternal;
+
+const logPrefix = "CreateServiceHandler";
+
+/**
+ * Type of a CreateService handler.
+ *
+ * CreateService expects a service payload as input
+ * and returns service with subscription keys
+ */
+type ICreateServiceHandler = (
+  context: Context,
+  auth: IAzureApiAuthorization,
+  clientIp: ClientIp,
+  attrs: IAzureUserAttributes,
+  servicePayload: ServicePayload
+) => Promise<ResponseTypes>;
+
+const createSubscriptionTask = (
+  logger: ILogger,
+  apiClient: ReturnType<APIClient>,
+  userEmail: EmailString,
+  subscriptionId: NonEmptyString,
+  productName: NonEmptyString
+): TaskEither<ErrorResponses, Subscription> =>
+  withApiRequestWrapper(
+    logger,
+    () =>
+      apiClient.createSubscription({
+        body: {
+          product_name: productName
+        },
+        email: userEmail,
+        subscription_id: subscriptionId
+      }),
+    200
+  );
+
+const createServiceTask = (
+  logger: ILogger,
+  apiClient: ReturnType<APIClient>,
+  servicePayload: ServicePayload,
+  subscriptionId: NonEmptyString,
+  sandboxFiscalCode: FiscalCode
+): TaskEither<ErrorResponses, Service> =>
+  withApiRequestWrapper(
+    logger,
+    () =>
+      apiClient.createService({
+        body: {
+          ...servicePayload,
```

Comment:
can you add a comment with a todo pointing to the relative task on pivotal to override token_name?

Reply:
Added in [5f377a4](https://github.com/pagopa/io-functions-services/pull/68/commits/5f377a49099c30bdca6dbaa2140ae01e448402f4)